test/integration/shoots/logging/utils.go

```diff
@@ -94,8 +94,8 @@ func WaitUntilLokiReceivesLogs(ctx context.Context, interval time.Duration, f *f
 	})
 
 	if err != nil {
-		d := time.Now().Add(5 * time.Minute)
-		dumpLogsCtx, dumpLogsCancel := context.WithDeadline(context.Background(), d)
+		// ctx might have been cancelled already, make sure we still dump logs, so use context.Background()
```

Comment:
> // ctx might have been cancelled already, make sure we still dump logs, so use context.Background()
Can you explain why this is a valid use case? Shouldn't it be the responsibility of the caller to pass a running context and if the passed context is cancelled, then shouldn't dumping the logs also stop?

Reply:
The caller (in this case the integration tests) passes a context with a timeout (overall test timeout). So once this is cancelled, we want to make sure the cleanup actions (e.g. dumping logs for further analysis) are started with a running context.
Of course, this could also be done in the `AfterTest` scope. However, I didn't want to refactor the whole test for this. At the end it's still non-production code.